File: tauri/apps/desktop/src/ipc/bot_settings.ts
/**
 * Bot settings IPC (plan 6.3) — persona, context window, AI tool policy.
 *
 * These go through the runtime operation, not a raw config write: the
 * operation validates the tool allow-list against the connector registry
 * and hot-swaps the live settings, so a change lands on the next message
 * instead of the next restart.
 *
 * Keep in sync with `tauri/apps/desktop/src-tauri/src/commands/bot_settings.rs`.
 */

import type { BotSettingsValue } from "@springtale/ui";
import { invoke } from "@tauri-apps/api/core";

export async function getBotSettings(): Promise<BotSettingsValue> {
  return invoke<BotSettingsValue>("get_bot_settings");
}

export async function saveBotSettings(settings: BotSettingsValue): Promise<void> {
  await invoke("save_bot_settings", { settings });
}

File: tauri/apps/desktop/src-tauri/src/commands/bot_settings.rs
//! Bot settings commands (plan 6.3) — persona, context window and the AI
//! tool allow-list, edited from the settings panel instead of a TOML file.

use serde::{Deserialize, Serialize};
use specta::Type;
use tauri::State;
use crate::runtime_guard::require_runtime;
use crate::state::AppState;

use springtale_runtime::operations::bot_settings::BotSettings;

/// Read the current bot settings (defaults when never saved).
#[tauri::command]
pub async fn get_bot_settings(state: State<'_, AppState>) -> Result<BotSettings, String> {
    let guard = require_runtime(&state.runtime).await?;
    let rt = guard.as_ref().unwrap();
    springtale_runtime::operations::bot_settings::get(&*rt.store)
        .await
        .map_err(|e| e.to_string())
}

/// Save bot settings. The operation validates every literal tool in the
/// allow-list against the connector registry before it writes.
#[tauri::command]
pub async fn save_bot_settings(
    state: State<'_, AppState>,
    settings: BotSettings,
) -> Result<(), String> {
    let guard = require_runtime(&state.runtime).await?;
    let rt = guard.as_ref().unwrap();
    springtale_runtime::operations::bot_settings::set(rt, settings)
        .await
        .map_err(|e| e.to_string())
}
